framework/db/Command.php

```diff
@@ -377,6 +377,11 @@ public function bindValues($values)
                 $this->pendingParams[$name] = [$value->getValue(), $value->getType()];
                 $this->params[$name] = $value->getValue();
             } else {
+                if (version_compare(PHP_VERSION, '8.1.0') >= 0
+                    && $value instanceof \UnitEnum
+                ) {
+                    $value = isset($value->value) ? $value->value : $value->name;
```

Comment:
Could we use '$value instanceof  \BackedEnum` here?

Reply:
This is a universal solution so you don't have to check for BackedEnum separately.

```php
$command->setSql('SELECT :p1')->bindValues([':p1' => enums\Status::ACTIVE]);
$this->assertSame('ACTIVE', $command->params[':p1']);

$command->setSql('SELECT :p1')->bindValues([':p1' => enums\StatusTypeString::ACTIVE]);
$this->assertSame('active', $command->params[':p1']);

$command->setSql('SELECT :p1')->bindValues([':p1' => enums\StatusTypeInt::ACTIVE]);
$this->assertSame(1, $command->params[':p1']);
```

Reply:
I've been a bit unclear. What I meant was to replace the `isset` check:
`$value = $value instanceof \BackedEnum ? $value->value : $value->name;`

Reply:
What do you think of this solution?

```php
if (version_compare(PHP_VERSION, '8.1.0') >= 0) {
    if ($value instanceof \BackedEnum) {
        $value = $value->value;
    } elseif ($value instanceof \UnitEnum) {
        $value = $value->name;
    }
}
````

Reply:
Added a correction.